chrome/content/zotero/itemTree.jsx

```diff
@@ -3822,7 +3847,11 @@ var ItemTreeRow = function(ref, level, isOpen)
 
 ItemTreeRow.prototype.getField = function(field, unformatted)
 {
-	return this.ref.getField(field, unformatted, true);
+	try {
+		return this.ref.getField(field, unformatted, true);
+	} catch (e) {
```

Comment:
We shouldn't use `catch` for this. If `getField()` throws, that's an error and we need to log/throw it. Maybe we need to add a flag when registering columns that says that it's not an item field and that `getCustomCellData()` needs to be used?

Reply:
Yeah, we shouldn't be using a catch here and likely should edit `_getRowData()` to call `Zotero.ItemTreeManager.getCustomCellData` for custom columns.

Reply:
`_getRowData` is a method of item tree, while I suppose getting data for row/cell from a custom source is a more high-level behavior that might also be applicable to other trees like the collection tree. Would it not be a good idea if we bundle this so deep inside item tree implementation? Something like the `getRowRenderer` might be a better place to call the `getCustomCellData`.

Reply:
Done

Reply:
> getting data for row/cell from a custom source is a more high-level behavior that might also be applicable to other trees like the collection tree.

It's only applicable to tabular trees that would get extended by plugins, which in Zotero is only the ItemTree and is unlikely in any reasonable future to be any other trees. The `getField()` method is part of the `ItemTreeRow` class, which itself suggests that it's specialized for ItemTree rows only, and the ItemTreeRow is an auxiliary class that is barely populated and has questionable reasons to exist at all.

> Something like the getRowRenderer might be a better place to call the getCustomCellData

VirtualizedTable accepts a custom rendering function precisely because you may provide whatever rendering function you want and pull data from wherever you want. `makeRowRenderer()` is just a reasonable default.

`_getRowData` is the correct place to get all column data for ItemTree rows, and also where you would reasonably perform a custom column check, since that's where all columns are iterated over and available for inspection. I don't think we need to worry about custom columns overriding Zotero's own fields, because the column registration code is preventing duplicates already.